src/main/java/uk/gov/companieshouse/web/accounts/controller/smallfull/DirectorsReportApprovalController.java

```diff
@@ -68,22 +69,19 @@ public String getDirectorsReportApproval(@PathVariable String companyNumber,
             DirectorsReportApproval directorsReportApproval =
                     directorsReportApprovalService.getDirectorsReportApproval(transactionId, companyAccountsId);
 
-            List<String> approverOptions =
-                    Arrays.stream(directorService.getAllDirectors(transactionId, companyAccountsId)).map(
-                            dir -> {
-
-                                if(dir.getResignationDate() == null) {
-                                    return dir;
-                                }
-                                else if (dir.getAppointmentDate().isAfter(dir.getResignationDate())) {
-                                    dir.setResignationDate(null);
-                                    return dir;
-                                }
-                                return dir;
-                            }
-                    ).filter(d -> d.getResignationDate() == null)
-                                    .map(Director::getName)
-                                            .collect(Collectors.toList());
+            List<String> approverOptions = new ArrayList<>();
+
+            for(Director director : directorService.getAllDirectors(transactionId, companyAccountsId)) {
+
+                if (director.getResignationDate() == null) {
+                    approverOptions.add(director.getName());
+                } else if (director.getAppointmentDate() != null) {
+                    if (director.getAppointmentDate().isAfter(director.getResignationDate())) {
+                        director.setResignationDate(null);
```

Comment:
This line isn't needed since `director` isn't used after this logic (besids to get the name)